### PR TITLE
cellSaveData: Log param error code

### PR DESCRIPTION
**Pull request by elad335**
May be useful for debugging.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 1]
```diff
@@ -318,68 +318,68 @@ static bool savedata_check_args(u32 operation, u32 version, vm::cptr<char> dirNa
...
 static NEVER_INLINE error_code savedata_op(ppu_thread& ppu, u32 operation, u32 version, vm::cptr<char> dirName,
 	u32 errDialog, PSetList setList, PSetBuf setBuf, PFuncList funcList, PFuncFixed funcFixed, PFuncStat funcStat,
 	PFuncFile funcFile, u32 container, u32 unk_op_flags /*TODO*/, vm::ptr<void> userdata, u32 userId, PFuncDone funcDone)
 {
-	if (!savedata_check_args(operation, version, dirName, errDialog, setList, setBuf, funcList, funcFixed, funcStat,
+	if (const auto ecode = savedata_check_args(operation, version, dirName, errDialog, setList, setBuf, funcList, funcFixed, funcStat,
 	funcFile, container, unk_op_flags, userdata, userId, funcDone))
 	{
-		return CELL_SAVEDATA_ERROR_PARAM;
+		return {CELL_SAVEDATA_ERROR_PARAM, std::to_string(ecode).c_str()};
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: It supports printable types already

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: I want it decimal though.

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 2]
```diff
@@ -988,7 +990,7 @@ static NEVER_INLINE error_code savedata_op(ppu_thread& ppu, u32 operation, u32 v
 				if (resv)
 				{
 					// ****** sysutil savedata parameter error : 59 ******
-					return CELL_SAVEDATA_ERROR_PARAM;
+					return {CELL_SAVEDATA_ERROR_PARAM, "19"};
```
Comment (jjsat) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Should this be "59"?

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Yes, thx.

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 3]
```diff
@@ -224,90 +224,90 @@ static error_code select_and_delete(ppu_thread& ppu)
 	return CELL_CANCEL;
 }
 
-static bool savedata_check_args(u32 operation, u32 version, vm::cptr<char> dirName,
+static s32 savedata_check_args(u32 operation, u32 version, vm::cptr<char> dirName,
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: wouldn't this be u32 ?

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: CELL_OK is s32.

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 4]
```diff
@@ -645,7 +646,7 @@ static NEVER_INLINE error_code savedata_op(ppu_thread& ppu, u32 operation, u32 v
 				{
 					doneGet->sizeKB += static_cast<s32>(::align(file.size, 4096));
 
-					if (!fs::remove_file(del_path + file.name))
+					if (!vfs::host::unlink(del_path + file.name))
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Don't use this unlink function. There is no assumption that remove_file might try to delete a file in use.

**Pull request merged**